tests/python_client/testcases/test_search.py

```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
+                                  100, default_search_exp)[0]
+        assert len(res[0]) == 100
+
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.xfail(reason="issue #23989")
+    def test_load_partition_drop_partition_delete(self):
+        """
+        target: test load partition drop partition delete
+        method: 1. create a collection and 2 partitions
+                2. insert data
+                3. load one partition
+                4. release and drop the partition
+                5. delete half data in each partition
+                6. search on the collection -> len(res)==50
+        expected: No exception
+        """
+        # insert data
+        collection_w = self.init_collection_general(prefix, partition_num=1)[0]
+        partition_w1, partition_w2 = collection_w.partitions
+        df = cf.gen_default_dataframe_data(200)
+        partition_w1.insert(df[:100])
+        partition_w2.insert(df[100:])
+        # load && release
+        partition_w2.load()
+        partition_w2.release()
+        partition_w2.drop()
+        # delete data
+        delete_ids = [i for i in range(50, 150)]
+        collection_w.delete(f"int64 in {delete_ids}")
+        # search
+        collection_w.load()
+        res = collection_w.search(vectors[:1], default_search_field, default_search_params,
+                                  100, default_search_exp)[0]
+        assert len(res[0]) == 100
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_compact_load_collection_release_partition(self):
+        """
+        target: test compact load collection release partition
+        method: 1. create a collection and 2 partitions
+                2. insert data multi times
+                3. compact
+                4. load the collection
+                5. release one partition
+                6. search on the released partition -> raise exception
+                7. search on the non-released partition
+        expected: No exception
+        """
+        collection_w = self.init_collection_general(prefix, partition_num=1)[0]
+        partition_w1, partition_w2 = collection_w.partitions
+        df = cf.gen_default_dataframe_data()
+        # insert data
+        partition_w1.insert(df[:100])
+        partition_w1.insert(df[100:200])
+        partition_w2.insert(df[200:300])
+        # compact
+        collection_w.compact()
+        collection_w.get_compaction_state()
+        # load && release
+        collection_w.load()
+        partition_w1.release()
+        # search
+        collection_w.search(vectors[:1], default_search_field, default_search_params,
+                            default_limit, default_search_exp, partition_names=[partition_w1.name],
+                            check_task=CheckTasks.err_res,
+                            check_items={ct.err_code: 1, ct.err_msg: 'not loaded'})
+        res = collection_w.search(vectors[:1], default_search_field, default_search_params,
```

Comment:
shall we add a search in the whole collection and check the results persist with results search in partition2